packages/react-dom/src/server/ReactPartialRenderer.js

```diff
@@ -692,8 +692,8 @@ class ReactDOMServerRenderer {
     // find the correct previous provider based on type
     let previousProvider;
     if (this.providerIndex > -1) {
-      for (let i = 0; i <= this.providerIndex; i += 1) {
-        if (this.providerStack[i] &&
+      for (let i = this.providerIndex; i >= 0; i -= 1) {
+        if (this.providerStack[i] !== null && this.providerStack[i] !== undefined &&
```

Comment:
So, do we actually need those checks? I wouldn't expect a provider "below" to ever be `null`. We only clear them out as we pop. As long as we check `i` doesn't get below zero we should be fine.

Reply:
Agreed. But, Flow doesn't seem to be aware of that. Perhaps we need some special Flow syntax I'm not aware of, which is highly likely?

Reply:
I addressed this by changing the array definition to be stricter and pop via setting the length. It didn't seem right to allow `null` as an array item but then not test for `null` when iterating over the array, even though the logic of how the array was being manipulated was OK.